code/game/objects/items/food/pizza.dm

```diff
@@ -16,11 +16,102 @@
 	venue_value = FOOD_PRICE_CHEAP
 	crafting_complexity = FOOD_COMPLEXITY_2
 	/// type is spawned 6 at a time and replaces this pizza when processed by cutting tool
-	var/obj/item/food/pizzaslice/slice_type
-	slice_type = /obj/item/food/pizzaslice
+	var/obj/item/food/pizzaslice/slice_type = /obj/item/food/pizzaslice
 	///What label pizza boxes use if this pizza spawns in them.
 	var/boxtag = ""
+	/// how many slices left
+	var/slices_left = 6
+	/// have we been sliced? like sliced and you can take it apart by hand
+	var/sliced = FALSE
+	/// cutting tools
+	var/static/list/cutting_tools = list(TOOL_KNIFE, TOOL_SAW, TOOL_SCALPEL)
+
+/obj/item/food/pizza/Initialize(mapload)
+	. = ..()
+	AddElement(/datum/element/drag_pickup)
+	register_context()
+
+/obj/item/food/pizza/add_context(atom/source, list/context, obj/item/held_item, mob/user)
+	. = ..()
+
+	if(istype(held_item) && (held_item.tool_behaviour in cutting_tools))
+		context[SCREENTIP_CONTEXT_LMB] = "Slice"
+		context[SCREENTIP_CONTEXT_RMB] = "Slice apart"
+		return TRUE
+
+	if(sliced)
+		context[SCREENTIP_CONTEXT_LMB] = "Take Slice"
+		return TRUE
+
+/obj/item/food/pizza/examine(mob/user)
+	. = ..()
+	if(isnull(slice_type) || !sliced)
+		return
+	. += span_notice("You can slice this to make it possible to take out slices with an empty hand!")
+
+/obj/item/food/pizza/item_interaction(mob/living/user, obj/item/tool, list/modifiers)
+	. = NONE
+	if(isnull(slice_type) || !(tool.tool_behaviour in cutting_tools))
+		return
+	if(!sliced)
+		slice(user, tool)
+		return ITEM_INTERACT_SUCCESS
+	user.visible_message(span_notice("[user] seperates [src] into individual slices with [tool]."))
+	cut_apart()
+	return ITEM_INTERACT_SUCCESS
+
+/obj/item/food/pizza/item_interaction_secondary(mob/living/user, obj/item/tool, list/modifiers)
+	. = NONE
+	if(isnull(slice_type) || !(tool.tool_behaviour in cutting_tools))
+		return
+	visible_message(span_notice("[user] seperates [src] into individual slices with [tool]."))
+	cut_apart()
+	return ITEM_INTERACT_SUCCESS
+
+/obj/item/food/pizza/attack_hand(mob/user, list/modifiers)
+	. = ..()
+	if(!sliced)
+		return
+	user.visible_message(span_notice("You take a slice of [src]."), span_notice("[user] takes a slice of [src]."))
+	produce_slice(user)
+
+/obj/item/food/pizza/proc/get_slices_filter() //to not repeat code
+	return alpha_mask_filter(icon = icon('icons/obj/food/pizza.dmi', "[slices_left]slices"))
 
+/// makes us possible to take slices out of
+/obj/item/food/pizza/proc/slice(mob/user, obj/item/tool)
+	if(sliced)
+		return
+	tool?.play_tool_sound(src)
+	sliced = TRUE
+	user?.visible_message(span_notice("[user] cuts [src] into 6 slices with [tool]."))
```

Comment:
If the args are optional they should be marked as such in the doc
If they aren't optional they need to error out when null